File: 2361-calculate-digit-sum-of-a-string/calculate-digit-sum-of-a-string.cs
public class Solution {
    public string DigitSum(string s, int k) {
      while(s.Length>k){
        string temp="";
        for(int i=0;i<s.Length;i+=k){
            int sum=0;
            for(int j=i;j<i+k && j<s.Length;j++){
                sum+=s[j]-'0';
            }
            temp+=sum.ToString();
        }
        s=temp;
      }
      return s;
    }}

File: 2361-calculate-digit-sum-of-a-string/calculate-digit-sum-of-a-string.py
class Solution:
    def digitSum(self, s: str, k: int) -> str:
        while(len(s)>k):
            word=[]
            for i in range(0,len(s),k):
                sum=0
                for j in range(i,min(i+k, len(s))):
                    sum+=int(s[j])
                word.append(str(sum))
            s="".join(word)
        return s
                
            

        